### PR TITLE
agent: Make intent of signaling channels clear and optimize memory

### PR DESCRIPTION
**Pull request by aditighag**
When channels are merely used for signaling purposes, use an empty
struct as the channel type since the value of the channel is never read.
Also, this can help with memory optimizations as the empty struct occupies 0 bytes of storage.

Signed-off-by: Aditi Ghag <aditi@cilium.io>

### CODE REVIEW

**Review by christarazi — APPROVED** [review]
:rocket:

**Comment by christarazi** [comment]
test-me-please

**Review by kkourt — APPROVED** [review]

**Comment by aditighag** [comment]
test-me-please

**Comment by aditighag** [comment]
test-me-please

File: pkg/health/server/prober.go [hunk 1]
```diff
@@ -43,8 +43,8 @@ type prober struct {
 
 	// 'stop' is closed upon a call to prober.Stop(). When the stopping is
 	// finished, then prober.Done() will be notified.
-	stop         chan bool
-	proberExited chan bool
+	stop         chan struct{}
+	proberExited chan struct{}
 	done         chan bool
```
Comment (tklauser) [inline pkg/health/server/prober.go]: I think the `done` channel could be changed to `chan struct{}` as well, though it requires changing `(*prober).Done()`.

Upon closer inspection this func even looks unused, so it doesn't look like anything is checking the channel being closed, so maybe we don't need the `done` channel after all.

**Review by tklauser — APPROVED** [review]
Nice change. One comment inline which could be addressed in a follow-up PR.

**Review by gandro — APPROVED** [review]
I agree with https://github.com/cilium/cilium/pull/14075#discussion_r526698797

We can probably just remove the `prober.Done()` function and `prober.done` channel in `pkg/health` to futher clean up the code.

The existing code waits for `prober.Stop()` to return and does not leak the `prober`, so there is really no use for `prober.Done()` as far as I can tell.

Not blocking the PR on this however.

File: pkg/health/server/prober.go (hunk 1 above)
Comment (aditighag) [reply pkg/health/server/prober.go]: Thanks...Done!

**Comment by aditighag** [comment]
test-me-please

**Comment by aditighag** [comment]
test-me-please

**Comment by aditighag** [comment]
test-me-please

**Comment by aditighag** [comment]
> I agree with [#14075 (comment)](https://github.com/cilium/cilium/pull/14075#discussion_r526698797)
> 
> We can probably just remove the `prober.Done()` function and `prober.done` channel in `pkg/health` to futher clean up the code.
> 
> The existing code waits for `prober.Stop()` to return and does not leak the `prober`, so there is really no use for `prober.Done()` as far as I can tell.
> 
> Not blocking the PR on this however.

Done

**Comment by jibi** [comment]
test-me-please

**Comment by jibi** [comment]
I (wrongly) restarted all tests as there was one (travis CI) stuck in _waiting for status to be reported_.
Turns out that test was already green https://travis-ci.com/github/cilium/cilium/builds/202619473
So I'm going ahead and merging this

**Pull request merged**